DolbyIO/Source/Private/Video/DolbyIOVideoTexture.cpp

```diff
@@ -109,12 +109,21 @@ namespace DolbyIO
 		    [SharedThis = AsShared()](FRHICommandListImmediate& RHICmdList)
 		    {
 			    FScopeLock Lock{SharedThis->GetBufferLock()};
-			    auto FRHITexture2D_Ptr = SharedThis->Texture->GetResource()->GetTexture2DRHI();
-			    uint32 SizeX = FRHITexture2D_Ptr->GetSizeX(), SizeY = FRHITexture2D_Ptr->GetSizeY();	
-			    RHIUpdateTexture2D(FRHITexture2D_Ptr, 0,
-			                       FUpdateTextureRegion2D{0, 0, 0, 0, SizeX, SizeY},
-			                       SizeX * Stride, SharedThis->GetBuffer());
-		    });
+
+				const auto TextureRegion = FUpdateTextureRegion2D{0, 0, 0, 0,
+																	static_cast<uint32>(SharedThis->Texture->GetResource()->GetTexture2DRHI()->GetSizeX()),
+																	static_cast<uint32>(SharedThis->Texture->GetResource()->GetTexture2DRHI()->GetSizeY())};
+
+				if (TextureRegion.DestX + TextureRegion.Width <= static_cast<uint32>(SharedThis->Texture->GetResource()->GetTexture2DRHI()->GetSizeX()) &&
```

Comment:
I think this is redundant is it not? DestX is 0 because you have created the region with 0 and Width will be SizeX() of the RHI texture as seen in the constructor. [FUpdateTextureRegion2D](https://docs.unrealengine.com/4.27/en-US/API/Runtime/RHI/FUpdateTextureRegion2D/__ctor/2/) is created with. Similarly for height/sizeY(). Thus I think this issue was already fixed with what was there previously in this [commit](https://github.com/DolbyIO/comms-sdk-unreal/commit/7fdc061934cb275db988e6082797ba415e7a1002) following your suggestion where it was made to use the RHI textures for size:
```
FScopeLock Lock{SharedThis->GetBufferLock()};
auto FRHITexture2D_Ptr = SharedThis->Texture->GetResource()->GetTexture2DRHI();
uint32 SizeX = FRHITexture2D_Ptr->GetSizeX(), SizeY = FRHITexture2D_Ptr->GetSizeY();	
RHIUpdateTexture2D(FRHITexture2D_Ptr, 0,
			                       FUpdateTextureRegion2D{0, 0, 0, 0, SizeX, SizeY},
			                       SizeX * Stride, SharedThis->GetBuffer());
```

Reply:
Yes that's right, I didn't see the change